### PR TITLE
Use the actual lockfile name in messages instead of assuming its alwa…

### PR DESCRIPTION
**Pull request by segiddins**
…ys Gemfile.lock

Closes https://github.com/bundler/bundler/issues/3672.

### CODE REVIEW

File: lib/bundler/definition.rb [hunk 1]
```diff
@@ -267,7 +267,7 @@ def lock(file)
       File.open(file, 'wb'){|f| f.puts(contents) }
     rescue Errno::EACCES
       raise Bundler::InstallError,
-        "There was an error while trying to write to Gemfile.lock. It is likely that \n" \
+        "There was an error while trying to write to #{File.basename(file)}. It is likely that \n" \
```
Comment (adarsh) [inline lib/bundler/definition.rb]: Why is this change different from the others? Shouldn't this be `#{Bundler.default_lockfile}`?

**Comment by adarsh** [comment]
This seems like a positive change and good contribution, however Travis is failing.

File: lib/bundler/definition.rb (hunk 1 above)
Comment (segiddins) [inline lib/bundler/definition.rb]: Why rely on global state when we have the proper file name handy?

Comment (adarsh) [inline lib/bundler/definition.rb]: Oh I see, this is the only place where we have `File` available so why not use it? I get that.

I lean slightly more on the side of consistency - if I were to see this break in the pattern, I'd wonder why, but I'm like a 2 out of 10 on this perspective.

@indirect thoughts? For me, this is ready to ship aside from this discussion.

**Comment by segiddins** [comment]
Travis should pass now, will merge when green

File: lib/bundler/definition.rb (hunk 1 above)
Comment (indirect) [inline lib/bundler/definition.rb]: I'm fine with using the local when it's available—avoiding globals when possible is a slightly stronger preference for me than perfect consistency.

**Comment by indirect** [comment]
:+1: :ship:

**Pull request merged**